packages/rooks/src/hooks/useIntersectionObserverRef.ts

```diff
@@ -26,10 +26,24 @@ function useIntersectionObserverRef(
 
   const [node, setNode] = useState<HTMLElementOrNull>(null);
 
+  const callbackRef = useRef(callback);
+  callbackRef.current = callback;
```

Comment:
This has to be updated in a separate useEffect

```suggestion
useEffect(() => {
    callbackRef.current = callback;
})
```

Reply:
Thanks for your review!